metadata-ingestion/src/datahub/ingestion/source/s3/config.py

```diff
@@ -64,23 +60,25 @@ class DataLakeSourceConfig(PlatformSourceConfigBase, EnvBasedSourceConfigBase):
         description="Maximum number of rows to use when inferring schemas for TSV and CSV files.",
     )
 
-    @pydantic.root_validator(pre=False)
-    def validate_platform(cls, values: Dict) -> Dict:
-        if not values.get("path_specs") and not values.get("path_spec"):
-            raise ValueError("Either path_specs or path_spec needs to be specified")
-
-        if values.get("path_specs") and values.get("path_spec"):
-            raise ValueError(
-                "Either path_specs or path_spec needs to be specified but not both"
-            )
-
-        if values.get("path_spec"):
-            logger.warning(
-                "path_spec config property is deprecated, please use path_specs instead of it."
-            )
-            values["path_specs"] = [values.get("path_spec")]
+    @pydantic.validator("path_specs", pre=True, always=True)
+    def rename_path_spec_to_path_specs(
+        cls, v: Any, values: Dict[str, Any]
+    ) -> List[PathSpec]:
+        """
+        Support the old path_spec field.
+        """
+        if "path_spec" in values:
+            assert (
+                "path_specs" not in values
+            ), 'cannot have both "path_spec" and "path_specs"'
+            logger.warning("path_spec is deprecated. Please use path_specs instead.")
+            return [values.pop("path_spec")]
+        return v
 
+    @pydantic.root_validator(pre=False)
+    def check_path_specs_and_infer_platform(cls, values: Dict) -> Dict:
         bucket_name: str = ""
+        path_spec: PathSpec
```

Comment:
should the code below (lines 93-95 in new) be replaced with?
```python
            if values.get("platform", ""):
                if values["platform"] != platform:
                    raise ValueError("all path_spec should belong to the same platform")
```

Reply:
yup changed